src/App.tsx

```diff
@@ -1,61 +1,43 @@
+import React, { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { TodoForm } from './components/TodoForm/TodoForm';
+import { User } from './types/User';
+
+const users: User[] = [...usersFromServer];
+
+function getUser(userId: number) {
+  return users.find(user => user.id === userId) || null;
+}
+
+const todosWithUser: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+
+  const handleAddTodo = (title: string, userId: number) => {
+    const newTodo = {
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+      title,
+      userId,
+      completed: false,
+      user: getUser(userId),
+    };
+
+    setTodos(prevTodos => [...prevTodos, newTodo]);
+  };
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
-
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      <TodoForm onAddTodo={handleAddTodo} />
```

Comment:
Ensure that the `TodoForm` component is receiving any necessary data as props, rather than importing directly from the server. This will help maintain a consistent data flow and make the component more reusable and testable.